tests/foreman/api/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,288 @@
+        assert compresource.name == new_cr_name
+        assert compresource.description == description
+
+        # Delete CR
+        compresource.delete()
+        assert not entities.AzureRMComputeResource().search(
+            query={'search': 'name={}'.format(new_cr_name)})
+
+    @upgrade
+    @tier2
+    def test_positive_create_finish_template_image(self,
+                                                   module_architecture,
+                                                   module_azurerm_cr,
+                                                   module_azurerm_finishimg):
+        """ Finish template image along with username is being added in AzureRM CR
+
+        :id: 78facb19-4b27-454b-abc5-2c69c0a6c28a
+
+        :steps:
+            1. Create a AzureRM Compute Resource.
+            2. Add a finish template based image in it.
+
+        :expectedresults: Finish template image should be added in AzureRM CR along with username
+
+        :CaseImportance: Critical
+
+        :CaseLevel: Integration
+        """
+
+        assert module_azurerm_finishimg.architecture.id == module_architecture.id
+        assert module_azurerm_finishimg.compute_resource == module_azurerm_cr
+        assert module_azurerm_finishimg.username == settings.azurerm.username
+        assert module_azurerm_finishimg.uuid == AZURERM_RHEL7_FT_IMG_URN
+
+    @upgrade
+    @tier2
+    def test_positive_create_cloud_init_image(self,
+                                              module_azurerm_cloudimg,
+                                              module_azurerm_cr,
+                                              module_architecture):
+        """Cloud Init template image along with username is being added in AzureRM CR
+
+        :id: 05ea1b20-0dfe-4af3-b1b7-a82604aa1e79
+
+        :steps:
+            1. Create a AzureRM Compute Resource.
+            2. Add a Cloud init supported image in it.
+
+        :expectedresults: Cloud init image should be added in AzureRM CR along with username
+
+        :CaseLevel: Integration
+        """
+
+        assert module_azurerm_cloudimg.architecture.id == module_architecture.id
+        assert module_azurerm_cloudimg.compute_resource.id == module_azurerm_cr.id
+        assert module_azurerm_cloudimg.username == settings.azurerm.username
+        assert module_azurerm_cloudimg.uuid == AZURERM_RHEL7_UD_IMG_URN
+
+    @upgrade
+    @tier2
+    def test_positive_check_available_networks(self, azurermclient, module_azurerm_cr):
+        """Check networks from AzureRM CR are available to select during host provision.
+
+        :id: aee5d077-668e-4f4d-adee-6472f0e4ecaa
+
+        :expectedresults: All the networks from AzureRM CR should be available.
+
+        :CaseLevel: Integration
+        """
+
+        cr_nws = module_azurerm_cr.available_networks()
+        portal_nws = azurermclient.list_network()
+        assert len(portal_nws) == len(cr_nws['results'])
```

Comment:
Can we compare with actual network names instead of their length of them?

Reply:
no , as both end provided same network in different ways ..hence only idea to see the count here.